packages/vix-core/src/plugins/serverBuildPlugin/index.test.ts

```diff
@@ -0,0 +1,94 @@
+import { vi, describe, expect, test, afterEach } from 'vitest';
+import serverBuildPlugin, {PLUGIN_NAME} from './index';
+import buildServer from '../../buildServer';
+
+describe('Return As Plugin', () => {
+  test('Return Object Structure keys', () => {
+    const OBJ_KEYS = ['name', 'enforce', 'apply', 'config', 'closeBundle'];
+    const plugin = serverBuildPlugin({});
+    for (let key of OBJ_KEYS) {
+      expect(plugin.hasOwnProperty(key)).toBeTruthy();
+    }
+  });
+
+  test('Check name, enforce and apply are string type', () => {
+    const OBJ_KEYS = ['name', 'enforce', 'apply'];
+    const plugin = serverBuildPlugin({});
+    for (let key of OBJ_KEYS) {
+      expect(plugin[key]).toBeTypeOf('string');
+    }
+  });
+
+  test('Check config, and closeBunlde are function type', () => {
+    const OBJ_KEYS = ['config', 'closeBundle'];
+    const plugin = serverBuildPlugin({});
```

Comment:
all three of these line can sit in **beforeEach**
`const plugin = serverBuildPlugin({});`

Reply:
**plugin** should be declared outside **beforeEach**, otherwise **plugin** wouldn't be accessed by test.
And if plugin is declared outside **beforeEach**, **let** should be used for assigning value.

  let plugin: Plugin;
  
  beforeEach(() => {
    plugin = serverBuildPlugin({});
  });